src/Microsoft.AspNetCore.Sockets.Client.Http/HttpConnection.cs

```diff
@@ -162,11 +162,11 @@ public Task StartAsync()
 
                     await _startTcs.Task;
                     await _receiveLoopTask;
-
                     _logger.LogDebug("Draining event queue");
-                    await _eventQueue.Drain();
 
-                    _httpClient.Dispose();
+                    await Task.WhenAny(
```

Comment:
This line of code is unreadable and too long. We should be using async await and removing the continue with. The task returned from WhenAny won't throw anyways. We should just add the helper that @BrennanConroy added to avoid throwing exceptions for faulted tasks.

We should also discuss configuring options for the client. We might need a builder API or something to make it so we can compose options.

This should look like `Task.WhenAny(eventQueue.Drain().NoThrow(), Task.Delay(Timeout))`